shell.nix

```diff
@@ -1,26 +1,18 @@
-{ pkgs ? import (fetchGit {
-    url = https://github.com/dapphub/dapptools.git;
-    rev = "6943c76bfb8e0b1fce54c3d9bba6f0f7e50d2f5c";
-  }) {}
-, mcd-cli ? pkgs.callPackage (import (fetchGit {
-    url = https://github.com/makerdao/mcd-cli.git;
-    rev = "86842b49defa53301ac0019f7d5994859bb3e1e9";
-  })) {}
-, sethret ? (import (fetchGit {
-    url = https://github.com/icetan/sethret.git;
-    rev = "ef77915e2881011603491275f36b44bf2478b408";
-  }) {}).sethret
-}:
+{ pkgsSrc ? (import ./nix/pkgs.nix {}).pkgsSrc
+, pkgs ? (import ./nix/pkgs.nix { inherit pkgsSrc; }).pkgs
+}: with pkgs;
 
-pkgs.mkShell {
-  buildInputs = with pkgs; [
-    dapp ethsign seth mcd-cli sethret
-    bc jq coreutils findutils procps
+let
+  tdds = import ./. { inherit pkgs; };
+in mkShell {
+  buildInputs = tdds.bins ++ [
```

Comment:
Where is `bins` defined?

Reply:
It's part of the `passthru` set of the `makerScriptsPackage` derivation: https://github.com/makerdao/nixpkgs-pin/blob/master/script-builder.nix#L26

`bins` are the binary dependencies of the deploy scripts.

Reply:
Why define them in that other repo instead of here?

Reply:
Because they are common to deploy scripts which `makerScriptsPackage` is supposed to package.

Reply:
I see. I wonder then if it wouldn't be more obvious if they were an attribute of `pkgs`, since they're not really specific to `tdds`. Additionally, in the case where we're building more than one deploy script, it would be weird to only add e.g. `tdds.bin` to `buildInputs`.

WDYT?

Reply:
Could definitely add common bins to `pkgs`. The reason for adding `bins` to `passthru` is that some packages might use the `extraBins` argument which adds non common bins.

Another solution would be to set `bins` to `propagatedBuildInputs` so that they are available by just adding `tdds` to `shell.nix`. This might solve your later statement also.